apps/antalmanac/src/components/Calendar/CalendarRoot.tsx

```diff
@@ -80,14 +80,15 @@ export default function ScheduleCalendar(props: ScheduleCalendarProps) {
     const [scheduleNames, setScheduleNames] = useState(AppStore.getScheduleNames());
 
     const { isMilitaryTime } = useTimeFormatStore();
-    const { hoveredCourseEvents } = useHoveredStore();
+    const { hoveredCalendarizedCourses, hoveredCalendarizedFinal } = useHoveredStore();
```

Comment:
Slice the store to reduce the number of dependencies this component has (i.e. how often it needs to re-render due to variables in the store updating).

There are different ways to slice it. Here's an example of returning only the desired properties in an array, and destructuring it.
 
```suggestion
    const [ hoveredCalendarizedCourses, hoveredCalendarizedFinal ] = useHoveredStore(store => [store.hoveredCalendarizedCourses, store.hoveredCalendarizedFinal]);
```

Reply:
I have incorporated this change and tried to study it, but my ability in React is a bit shaky in understanding this usage regardless. Could you perhaps briefly explain your use context of the arrow function here (for the sake of my learning, haha)?

Thank you!

Reply:
Here's [Zustand's documentation on slicing](https://docs.pmnd.rs/zustand/guides/slices-pattern)

I visualize it like this:

```tsx
const useMyStore = () => {
  const [state1, setState1] = useState()
  const [state2, setState2] = useState()
  const [state3, setState3] = useState()
  
  return {
    state1, state2, state3, setState1, setState2, setState3
  }
}
```

Here, a custom hook provides you with 3 different stateful values. Any component that uses this hook, will re-render whenever any of those three state variables change (i.e. when their corresponding setter is called). Zustand's slicing pattern allows you to select which state variables to actually subscribe to via a callback.

```tsx
  const [state1, setState1] = useState()
  const [state2, setState2] = useState()
  const [state3, setState3] = useState()
  
  return sliceFunction({ state1, state2, state3, setState1, setState2, setState3 })
``` 

Whenever you provide a custom slicing function, it allows you to select a sub-set of stateful values to actually use in the component, so you can reduce the number of renders. This visualization isn't entirely accurate to reality, but it's how I interpret it 🤓

Reply:
Thank you! Very helpful!!